org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/M2EUIPluginActivator.java

```diff
@@ -180,4 +199,25 @@ public ArchetypePlugin getArchetypePlugin() {
     return this.archetypeManager.getService();
   }
 
+  private static void deleteLegacyCacheDirectory(File m2eCache) {
+    new UIJob("Delete legacy M2E cache") {
+      @Override
+      public IStatus runInUIThread(IProgressMonitor monitor) {
+        int selection = MessageDialog.open(MessageDialog.QUESTION, getDisplay().getActiveShell(),
+            "Delete obsolete M2E cache?",
+            "A cache directory used by previous M2E versions was detected:\n" + m2eCache + "\n\n"
+                + "It's no longer used by newer M2E versions and, unless older Eclipse installations need it, can be safely deleted.",
+            SWT.NONE, "Keep Cache", "Delete Cache");
```

Comment:
you probably need a don't ask again choice as well, else it'll nag users who decide to keep the cache on every start

Reply:
That's really a huge effort just for users switching back-and.forth between old and new m2e ... So probably simply let them waste theri disk-space than wasting development and maintenance effort here... @mickaelistria

Reply:
> That's really a huge effort just for users switching back-and.forth between old and new m2e ... So probably simply let them waste theri disk-space than wasting development and maintenance effort here... @mickaelistria

It was indeed not the intended to end up with that much code. :/


> I would also suggest storing the result globally for the user, so that they won't be asked for every new eclipse installation again.

They should at least be ask for each Eclipse installation, because otherwise it will probably forgotten if the last installation started to use M2E 2.

Reply:
Well, IMHO that's still too much in certain environments, where people deploy new eclipse installations weekly rather than yearly. With oomph, things are set up automatically, and that popup would interrupt the automatic installation process (git clone, project import, etc.). These modal dialogs on startup (takari stats, jboss usage and now this) are not helpful in that regard.

There could be a system property to disable the popup. Or make it a passive notification instead of a modal dialog.

Reply:
OK. I actually never wanted to make it that complicated, but before we get many complaints and since I had started it ...
You can now set the Java-system property `m2e.keep.legacy.cache` to `true` to just keep the cache.
Furthermore you can set a check-box to be not asked again.
In that case a marker file `DELETE_ME.txt` is created in the directory that explains why/when this can be deleted and that prevents the dialog from popping up subsequently.

If that is not sufficient I will just add the file to the directory so one can delete it manually if stumbling upon it.
This way maybe gigabytes of storage are kept unnecessarily occupied but at least no pop-up shows up.